github/ci/services/prometheus-stack/manifests/stack/common/kube-prometheus-stack.yaml

```diff
@@ -1592,3 +1592,28 @@ spec:
   endpoints:
   - port: web
     path: "/metrics"
+---
+# PodMonitor to scrape prometheus metrics from each ephemeral Prometheus instance of each prow-job kubevirtci cluster
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: prometheus-stack-federation
+  namespace: default
+  labels:
+    app: kube-prometheus-stack-prometheus-federation
+
+    release: "prometheus-stack"
+    group: kubevirtci
+spec:
+  podMetricsEndpoints:
+  - port: web
```

Comment:
The default prometheus service has the port called web.
The port field is the name of the port, and the old field targetPort is deprecated.
But looks like can hadcode the port number with
```
- sourceLabels:
        - __meta_kubernetes_pod_container_port_number
        action: keep
        regex: "30007"
```

Reply:
> The default prometheus service has the port called web.

Yes, the "default port" inside the kubevirtci cluster  which runs in the pod. This is not magically visible on the outside of the pod. And then having a port named `web` on the pod is not very understandable, because it is  not a "prometheus" pod. It is a pod with a whole cluster. Right?

Reply:
> The port field is the name of the port, and the old field targetPort is deprecated.

Yes deprecated but you can still use it. As I said, as an alternative you can also add a named port definition to each pod_spec on the port. Just don't call it `web`.